pkg/infrastructure/attribute_manager_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package infrastructure
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const Path string = "/tmp/taxonomy/infraattributes.json#/definitions/Infrastructure"
+
+func TestValidInfrastructureAttributeSC(t *testing.T) {
+	t.Parallel()
+
+	data := `{
+		"infrastructure":[{
+			"attribute": "storage-cost",
+			"description": "neverland object store",
+			"value": "100",
+			"type": "numeric",
+			"units": "US Dollar",
+			"instance": "account-neverland"
+		}]
+	}`
+	validateErr := validateStructure([]byte(data), Path)
+	assert.Nil(t, validateErr, "No error should be found")
+}
+
+func TestInvalidInfrastructureAttributeSC(t *testing.T) {
+	t.Parallel()
+
+	data := `{
+		"infrastructure":[{
+			"attribute": "storage-cost",
+			"description": "neverland object store",
+			"value": "100",
+			"type": "numeric",
+			"units": "USDollar",
+			"instance": "account-neverland"
+		}]
+	}`
+	validateErr := validateStructure([]byte(data), Path)
+	assert.Error(t, validateErr, "An error is expected")
+}
+
+func TestValidInfrastructureAttributeBW(t *testing.T) {
+	t.Parallel()
+
+	data := `{
+		"infrastructure":[{
+			"attribute": "bandwidth",
+			"description": "neverland object store",
+			"value": "100",
+			"type": "numeric",
+			"units": "GB/s",
+			"instance": "account-neverland"
+		}]
+	}`
+	validateErr := validateStructure([]byte(data), Path)
+	assert.Nil(t, validateErr, "No error should be found")
+}
+
+func TestInvalidInfrastructureAttributeBW(t *testing.T) {
+	t.Parallel()
+
+	data := `{
+		"infrastructure":[{
+			"attribute": "bandwidth",
+			"description": "neverland object store",
+			"value": "100",
+			"type": "numeric",
+			"units": "GB",
```

Comment:
@shlomitk1 , maybe this is unrelated to the PR, but do we have or plan to implement units transformation/comparison. I mean if we have 2 infrastructure entries one with 1GB and another 1000 (1024) MB, how they will be compared?

Reply:
We have but for the future. The first version of optimizer will assume one type of units per attribute.